pkg/cloudprovider/aws/fake/fake.go

```diff
@@ -0,0 +1,142 @@
+package fakeaws
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/autoscaling"
+	"github.com/aws/aws-sdk-go/service/ec2"
+
+	"github.com/aws/aws-sdk-go/service/autoscaling/autoscalingiface"
+	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+)
+
+var (
+	defaultAvailabilityZone = "us-east-1a"
+)
+
+type Instance struct {
+	InstanceID           string
+	AutoscalingGroupName string
+	State                string
+}
+
+type Ec2 struct {
+	ec2iface.EC2API
+
+	Instances *[]*Instance
```

Comment:
Related to my other comment on pointer to a slice - why?

Reply:
It was to take into terminating instances using the ec2 instance and then describing asgs using the autoscaling instance and I want them to be in sync. I've done some testing and moved to make to make it easier to manage those fake instances in the future and it works as I'd like.